### PR TITLE
More flexible consent backend

### PR DESCRIPTION
**Pull request by isoos**
This is in preparation the backend for unauthenticated access.

### CODE REVIEW

File: app/lib/account/consent_backend.dart
Comment (jonasfj) [inline app/lib/account/consent_backend.dart:251]: Why not simply `tx.delete(consent.key)` -- also should we not being doing this in the same transaction as everything else done when accepting the invite?

Like maybe the `tx` should be passed to `onAccept`

File: app/lib/account/consent_backend.dart
Comment (jonasfj) [inline app/lib/account/consent_backend.dart:265]: Why not simply `tx.delete(consent.key)`? I assume we would have to handle the case where it's already deleted.. but that's about it?

**Review by jonasfj — APPROVED** [review]
just things we should consider that doesn't block this PR :D

File: app/lib/account/consent_backend.dart
Comment (isoos) [reply app/lib/account/consent_backend.dart:251]: This code predates the retry-datastore-transactions code (hence the outer retry), and it was not merged since. I'll do a follow-up with that.

File: app/lib/account/consent_backend.dart
Comment (isoos) [reply app/lib/account/consent_backend.dart:265]: This is the consequence of the same outer retry block. This should be removed once the onaccept and the transaction is merged.

**Pull request merged**
